windows-msi/Makefile

```diff
@@ -0,0 +1,229 @@
+#
+#
+
+PRODUCT_PUBLISHER=OpenVPN Technologies, Inc.
+PRODUCT_NAME=OpenVPN
+PRODUCT_VERSION=2.4.6
+PRODUCT_VERSION_GUID={6094AE77-0BCA-407E-B30E-747E7AABCA81}
+CONFIG_EXTENSION=ovpn
+
+#
+# Folder with compiled OpenVPN
+# ("-x86_64" or "-i686" is appended)
+#
+!IFNDEF SOURCE_DIR
+SOURCE_DIR=image
+!ENDIF
+
+#
+# Folder with Easy RSA
+#
+!IFNDEF EASYRSA_DIR
+EASYRSA_DIR=image-easy-rsa
+!ENDIF
+
+#
+# Folder with openvpnserv2.exe
+#
+!IFNDEF OPENVPNSERV2_DIR
+OPENVPNSERV2_DIR=image-openvpnserv2
+!ENDIF
+
+#
+# Folder to store intermmediate files while building
+#
+!IFNDEF BUILD_DIR
+BUILD_DIR=build.tmp
+!ENDIF
+
+#
+# Folder to store output files
+#
+!IFNDEF OUTPUT_DIR
+OUTPUT_DIR=image
+!ENDIF
+
+#
+# Platform to build (i386 or amd64)
+#
+!IFNDEF PLAT
+PLAT=amd64
+!ENDIF
+
+#
+# Platform-specific parameters
+#
+!IF "$(PLAT)" == "i386"
+SOURCE_DIR=$(SOURCE_DIR)-i686
+PROGRAM_FILES_DIR=ProgramFilesFolder
+OPENSSL_PLAT=
+WIX_PLAT=x86
+PRODUCT_UPGRADE_GUID={30B0C64D-FEF6-4A4E-ABBA-750531E0F3FB}
+!ELSEIF "$(PLAT)" == "amd64"
+SOURCE_DIR=$(SOURCE_DIR)-x86_64
+PROGRAM_FILES_DIR=ProgramFiles64Folder
+OPENSSL_PLAT=-x64
+WIX_PLAT=x64
+PRODUCT_UPGRADE_GUID={E5B5C1A4-F5CE-4010-90DD-7984AC8B5FF2}
+!ELSE
+!ERROR Unknown platform. Please, set PLAT to "i386" or "amd64".
+!ENDIF
+
+
+#
+# WiX flags
+#
+WIX_CANDLE_FLAGS=$(WIX_CANDLE_FLAGS) \
+	-nologo \
+	-ext WixUtilExtension \
+	-arch "$(WIX_PLAT)" \
+	-dPRODUCT_PUBLISHER="$(PRODUCT_PUBLISHER)" \
+	-dPRODUCT_NAME="$(PRODUCT_NAME)" \
+	-dPRODUCT_VERSION="$(PRODUCT_VERSION)" \
+	-dPRODUCT_VERSION_GUID="$(PRODUCT_VERSION_GUID)" \
+	-dCONFIG_EXTENSION="$(CONFIG_EXTENSION)" \
+	-dPROGRAM_FILES_DIR="$(PROGRAM_FILES_DIR)" \
+	-dOPENSSL_PLAT="$(OPENSSL_PLAT)" \
+	-dPRODUCT_UPGRADE_GUID="$(PRODUCT_UPGRADE_GUID)"
+WIX_LIGHT_FLAGS=$(WIX_LIGHT_FLAGS) \
+	-nologo \
+	-dcl:high \
+	-spdb \
+	-ext WixUtilExtension \
+	-b build="$(BUILD_DIR)" \
+	-b openvpn="$(SOURCE_DIR)" \
+	-b easyrsa="$(EASYRSA_DIR)" \
+	-b openvpnserv2="$(OPENVPNSERV2_DIR)"
+
+
+#
+# Main Targets
+# (Default one first)
+#
+
+All ::
+
+Clean ::
+
+
+#
+# WiX Sources
+#
+
+"$(BUILD_DIR)\gui.wixobj" : "gui.wxs" "$(BUILD_DIR)"
+	"$(WIX)bin\candle.exe" $(WIX_CANDLE_FLAGS) -out $@ "gui.wxs"
+
+"$(BUILD_DIR)\$(PLAT)\msi.wixobj" : "msi.wxs" "$(BUILD_DIR)\$(PLAT)"
+	"$(WIX)bin\candle.exe" $(WIX_CANDLE_FLAGS) -out $@ "msi.wxs"
+
+"$(OUTPUT_DIR)\$(PRODUCT_NAME)-$(PLAT).msi" : \
+	"artwork\info.ico" \
+	"artwork\folder_new.ico" \
+	"artwork\folder_up.ico" \
+	"artwork\header.bmp" \
+	"artwork\openvpn.ico" \
+	"$(SOURCE_DIR)\bin\libcrypto-1_1$(OPENSSL_PLAT).dll" \
+	"$(SOURCE_DIR)\bin\liblzo2-2.dll" \
+	"$(SOURCE_DIR)\bin\libpkcs11-helper-1.dll" \
+	"$(SOURCE_DIR)\bin\libssl-1_1$(OPENSSL_PLAT).dll" \
+	"$(SOURCE_DIR)\bin\openssl.exe" \
+	"$(SOURCE_DIR)\bin\openvpn.exe" \
+	"$(SOURCE_DIR)\bin\openvpn-gui.exe" \
+	"$(SOURCE_DIR)\bin\openvpnserv.exe" \
+	"$(OPENVPNSERV2_DIR)\openvpnserv2.exe" \
+	"$(BUILD_DIR)\README-config.txt" \
+	"..\windows-nsis\INSTALL-win32.txt" \
```

Comment:
True. However, I can't just simply fork it to the `windows-msi` folder. Its content needs to be updated.

I suppose it would be best to update and move the building instructions to a Wiki page, and link to it from the `INSTALL-win32.txt` document. This document is installed to end-user computers and should focus on Windows-specific OpenVPN setup.

Reply:
@rozmansi I think we could move INSTALL-win32.txt to the root of the repository and, as you suggest, move the actual build instructions (NSIS/MSI) to the Trac wiki. The percentage of Windows users who actually cross-compile or package OpenVPN is so small that showing the instructions to everyone is rather pointless.

Reply:
Fixed in b127c568a04ca3f1ab3df430d1224dad489e47d2.